pdfannots/printer/csv.py

```diff
@@ -0,0 +1,38 @@
+import argparse
+import io
+import typing
+import csv
+
+from ..types import Document
+from . import DictBasedPrinter
+
+
+class CsvPrinter(DictBasedPrinter):
+    def __init__(self, args: argparse.Namespace):
+        super().__init__(args)
+        self.remove_hyphens = args.remove_hyphens  # Whether to remove hyphens across a line break
+
+        # Multiple input files in a single output require to have a filename
+        self.printfilename = args.printfilename if (
+            not (hasattr(args, "input") and len(args.input) > 1)) else True
+
+        self.memcsvfile = io.StringIO()
+        self.writer = csv.DictWriter(self.memcsvfile, fieldnames=(
+            ['filename'] if self.printfilename else []) + self.all_fieldnames)
```

Comment:
This looks fragile. I think we'd need the field names and order to be more stable than JSON for future compatibility.